cli.js

```diff
@@ -1,155 +1,157 @@
-      }
-    });
-  } else {
-    var meta=(argv.m).split('=');
-    if (meta.length === 2) {
-      metaElements[meta[0]] = meta[1];
-    }
-  }
-}
-
-logger.info('output directory: %s', outDir);
-if (target.isDirectory()) {
-  // the ajv json validator will be passed into the main module to help with processing
-  var files=[];
+#! /usr/bin/env node
+/**
+ * Copyright 2017 Adobe Systems Incorporated. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+var Promise = require('bluebird');
+var path = require('path');
+var _ = require('lodash');
+var fs = Promise.promisifyAll(require('fs'));
+var readdirp = require('readdirp');
+var Ajv = require('ajv');
+var winston = require('winston');
+
+var Schema = require('./lib/schema');
+var readSchemaFile = require('./lib/readSchemaFile');
+
+// parse/process command line arguments
+var argv = require('optimist')
+  .usage('Generate Markdown documentation from JSON Schema.\n\nUsage: $0')
+  .demand('d')
+  .alias('d', 'input')
+  // TODO: is baseURL still a valid parameter?
+  .describe('d', 'path to directory containing all JSON Schemas or a single JSON Schema file. This will be considered as the baseURL. By default only files ending in .schema.json will be processed, unless the schema-extension is set with the -e flag.')
+  .alias('o', 'out')
+  .describe('o', 'path to output directory')
+  .default('o', path.resolve(path.join('.', 'out')))
+  .alias('m', 'meta')
+  .describe('m', 'add metadata elements to .md files Eg -m template=reference. Multiple values can be added by repeating the flag Eg: -m template=reference -m hide-nav=true')
+  .alias('s', 'metaSchema')
+  .describe('s', 'Custom meta schema path to validate schemas')
+  .alias('x', 'schema-out')
+  .describe('x', 'output JSON Schema files including description and validated examples in the _new folder at output directory, or suppress with -')
+  .alias('e', 'schema-extension')
+  .describe('e', 'JSON Schema file extension eg. schema.json or json')
+  .alias('n', 'no-readme')
+  .describe('v', 'JSON Schema Draft version to use. Supported: 04, 06, 07 (default)')
+  .alias('v', 'draft')
+  .default('v', '07')
+  .describe('n', 'Do not generate a README.md file in the output directory')
+  .describe('link-*', 'Add this file as a link the explain the * attribute, e.g. --link-abstract=abstract.md')
+  .check(function(args) {
+    if (!fs.existsSync(args.input)) {
+      throw 'Input file "' + args.input + '" does not exist!';
+    }
+    if (args.s && !fs.existsSync(args.s)) {
+      throw 'Meta schema file "' + args.s + '" does not exist!';
+    }
+  })
+  .alias('p', 'properties')
+  .describe('p', 'A comma separated list with custom properties which should be also in the description of an element.')

```

Comment:
```suggestion
  .describe('p', 'A comma separated list with custom properties which should be also in the description of an element.')
  .array('p')
```

This will use the built-in [multi-value support in yargs](https://yargs.js.org/docs/#api-arraykey), so that you can call `jsonschema2md -p version -p test` or `jsonschema2md -p version test`. I think that's more robust than trying to split the array yourself. (You'd have to adjust the rest)

Reply:
@trieloff 
As far as I see the parser which is used did not have the function array.
![image](https://user-images.githubusercontent.com/31853210/62117545-66cfc900-b2bc-11e9-9070-8139984ad59a.png)

So your suggestion will result in an error 
![image](https://user-images.githubusercontent.com/31853210/62117668-9b438500-b2bc-11e9-8cb5-ef4515fe622f.png)

Reply:
My bad, I forgot we are using optimist, not yargs.